haystack/preview/components/prompt/models/base.py

```diff
@@ -0,0 +1,103 @@
+from typing import Type, Optional, Dict, Any, List
+
+import logging
+
+from canals import _find_decorated_classes
+
+
+logger = logging.getLogger(__name__)
+
+
+class PromptModelError(Exception):
+    pass
+
+
+def prompt_model(class_):
```

Comment:
I like the implementation, but I don't like the naming as the usage is not very clear to me.

Will this be only used as a decorator as shown in the tests? or validator?